plugins/Annotations/tests/System/AnnotationsTest.php

```diff
@@ -264,30 +264,33 @@ public function getPermissionsFailData()
      */
     public function testMethodPermissions($hasAdminAccess, $hasViewAccess, $request, $checkException, $failMessage)
     {
+        if (true === $checkException) {
+            self::expectException(Exception::class);
+        } else {
+            self::expectNotToPerformAssertions();
+        }
+
         // create fake access that denies user access
-        FakeAccess::$superUser = false;
+        FakeAccess::clearAccess(false);
+        FakeAccess::$identity = 'user' . (int)$hasAdminAccess . (int)$hasViewAccess;
         FakeAccess::$idSitesAdmin = $hasAdminAccess ? array(self::$fixture->idSite1) : array();
         FakeAccess::$idSitesView = $hasViewAccess ? array(self::$fixture->idSite1) : array();
 
-        if ($checkException) {
-            try {
-                $request = new Request($request);
-                $request->process();
-                $this->fail($failMessage);
-            } catch (Exception $ex) {
-                $this->assertTrue(true); // pass
-            }
-        } else {
-            $request = new Request($request);
-            $request->process();
-
-        }
+        $request = new Request($request.'&format=original');
+        $request->process();
```

Comment:
The requests actually did not contain a format, so xml response was returned. In case of an error there was no exception thrown by the request. But with the meanwhile used PHPUnit version `$this->fail()` throws an `AssertionFailedError` exception, which was directly caught and `$this->assertTrue(true)` did make all tests pass.

Will look through the other tests as well and check where we use `$this->fail()` in a try/catch